qiskit_ibm_provider/job/ibm_circuit_job.py

```diff
@@ -184,8 +193,8 @@ def properties(self) -> Optional[BackendProperties]:
             IBMJobApiError: If an unexpected error occurred when communicating
                 with the server.
         """
-        with api_to_job_error():
-            properties = self._api_client.job_properties(job_id=self.job_id())
+        with api_to_job_error():  # TODO: are those the correct properties?
```

Comment:
Should we raise an error (crashing all the code currently depending on this) or print warning and return `None`?

Also, have to admit I didn't understand what should be written in the error text.

Reply:
We can just delete the method and note in the migration guide that this is no longer supported. Since the user would get an error either way.

Same goes to the other methods that I said could raise an exception (e.g. job.name()). Sorry about the confusion.